spyderplugins/widgets/pylintgui.py

```diff
@@ -61,7 +61,8 @@ def get_pylint_version():
                                shell=True if os.name == 'nt' else False)
     lines = to_unicode_from_fs(process.stdout.read()).splitlines()
     if lines:
-        match = re.match('(pylint3?|pylint-script.py) ([0-9\.]*)', lines[0])
+        regex = '({0}|pylint-script.py) ([0-9\.]*)'.format(PYLINT)

```

Comment:
I understood "pylint3?" to mean to match either "pylint" or "pylint3".
PYLINT equals "pylint" or "pylint3" or "python3-pylint" depending on the
binary that was already previously found.

Was the previous behavior to allow the version string to be different from
the binary name?
- pylint but "pylint3 1.3.1"
- pylint3 but "pylint 1.3.1"
Does this happen in practice?

I originally used "(python3-pylint|pylint3?|pylint-script.py)" but thought
that it was a burden to have to modify the regex each time a new binary
name cropped up.

Or simpler, why not just skip the first word? The code already assumes the
version string is on the first line.
On 12 May 2015 05:23, "Carlos Cordoba" <notifications@github.com> wrote:

> In spyderplugins/widgets/pylintgui.py
> <https://github.com/spyder-ide/spyder/pull/2434#discussion_r30083813>:
>
> > @@ -61,7 +61,8 @@ def get_pylint_version():
> >                                 shell=True if os.name == 'nt' else False)
> >      lines = to_unicode_from_fs(process.stdout.read()).splitlines()
> >      if lines:
> > -        match = re.match('(pylint3?|pylint-script.py) ([0-9\.]*)', lines[0])
> > +        regex = '({0}|pylint-script.py) ([0-9\.]*)'.format(PYLINT)
>
> I think a ? is missing here after {0}, at least according to the changed
> line above :-)
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/spyder-ide/spyder/pull/2434/files#r30083813>.
>